pkg/manager/client/k8s.go

```diff
@@ -51,6 +51,26 @@ func (k *KubernetesClient) GetPodContainerLogRequest(namespace, podName, contain
 	})
 }
 
+func (k *KubernetesClient) GetPodContainerPreviousLogRequest(namespace, podName, containerName string) *rest.Request {
+	return k.clientSet.CoreV1().Pods(namespace).GetLogs(podName, &corev1.PodLogOptions{
+		Container:  containerName,
+		Timestamps: true,
+		Previous:   true,
+	})
+}
+
+func (k *KubernetesClient) GetPodRestartCount(namespace, podName, containerName string) (int32, error) {
+	pod, err := k.clientSet.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+	if err != nil {
+		return 0, err
+	}
+	if len(pod.Status.ContainerStatuses) > 0 {
```

Comment:
the current logic returns restart count for the first container in containerStatus. We should refactor this to make sure containerName in containerStatus matches the input containerName and fetch restart count accordingly. For example a sample pod container status:

```
  containerStatuses:
  - containerID: containerd://15028df52f985f91f2451ed2a51d3e6e26c23116729c535792d99e32e8d1cbc5
    image: docker.io/longhornio/longhorn-manager:v1.3.2
    imageID: sha256:de93f80351954b18b95039c7f2e473319a0e086f6b33e0402fcff654ee52933b
    lastState:
      terminated:
        containerID: containerd://717fe507a6a9ad6dcdb0f098e5c4f6a7f4a8c196be8cdf44dd0dfb527f358bf8
        exitCode: 1
        finishedAt: "2023-07-16T00:58:31Z"
        reason: Error
        startedAt: "2023-07-16T00:58:21Z"
    name: longhorn-csi-plugin
    ready: true
    restartCount: 7
    started: true
    state:
      running:
        startedAt: "2023-07-16T00:58:59Z"
  - containerID: containerd://3e7ef1f4c01767961fdc38cd1d861a0c267412ae1cd83e05daebcb10d97fa46b
    image: docker.io/longhornio/csi-node-driver-registrar:v2.5.0
    imageID: sha256:cb03930a2bd4247929205f328d7c1b7d9594d7586813d4c108c43dc852fad219
    lastState:
      terminated:
        containerID: containerd://2546ed7f27f25a5069c609e61bcea1a88502b10ff0aba2c6363c6bd9c06fc8ac
        exitCode: 255
        finishedAt: "2023-07-16T00:56:17Z"
        reason: Unknown
        startedAt: "2023-07-04T16:12:21Z"
    name: node-driver-registrar
    ready: true
    restartCount: 3
    started: true
    state:
      running:
        startedAt: "2023-07-16T00:57:57Z"
```

Reply:
Nice catch! Updated!